gen/gen.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"reflect"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type Data struct {
+	Name string
+	*Yml
+}
+
+func GenCHeader(data *Data, dst io.Writer) error {
+	t := template.
+		New("").
+		Funcs(template.FuncMap{
+			"SComment":     func(v string, indent int) string { return Comment(v, CommentTypeSingleLine, indent, true) },
+			"MComment":     func(v string, indent int) string { return Comment(v, CommentTypeMultiLine, indent, true) },
+			"SCommentN":    func(v string, indent int) string { return Comment(v, CommentTypeSingleLine, indent, false) },
+			"MCommentN":    func(v string, indent int) string { return Comment(v, CommentTypeMultiLine, indent, false) },
+			"ConstantCase": ConstantCase,
+			"PascalCase":   PascalCase,
+			"CamelCase":    CamelCase,
+			"CType":        CType,
+			"CValue":       CValue,
+			"BitFlagValue": func(index int) uint64 { return uint64(math.Pow(2, float64(index-1))) },
+			"ParseUint": func(s string, bitSize int) uint64 {
+				v, err := strconv.ParseUint(s, 10, bitSize)
+				if err != nil {
+					panic(err)
```

Comment:
Template functions can return an error, so you don't have to use panic. e.g.

`func(s string, bitSize int) uint64` -> `func(s string, bitSize int) (uint64, error)`